test-harness/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -30,18 +27,42 @@ public void invalidSchemaShouldNotSucceed() throws Exception {
     }
 
     @Test
-    public void testRetrieveDocStringFromAttribute() {
-        String expectedDocString = "If checked, this will allow users who are not authenticated to access Jenkins in a read-only mode.";
-        String actualDocString = retrieveDocStringFromAttribute(hudson.security.FullControlOnceLoggedInAuthorizationStrategy.class, "allowAnonymousRead");
-        assertEquals(expectedDocString,actualDocString);
+    public void rejectsInvalidBaseConfigurator() throws Exception {
+        assertThat(validateSchema(convertYamlFileToJson(this, "invalidBaseConfig.yml")),
+            contains("#: extraneous key [invalidBaseConfigurator] is not permitted"));
     }
 
     @Test
-    public void testRemoveHtmlTagRegex() {
-        String htmlTagString = "<div> If checked, this will allow users who are not authenticated to access Jenkins in a read-only mode.</div>";
-        String expectedString = "If checked, this will allow users who are not authenticated to access Jenkins in a read-only mode.";
-        String actualString = removeHtmlTags(htmlTagString);
-        assertEquals(expectedString, actualString);
+    public void validJenkinsBaseConfigurator() throws Exception {
+        assertThat(validateSchema(convertYamlFileToJson(this, "validJenkinsBaseConfig.yml")),
+            empty());
     }
 
+    @Test
+    public void symbolResolutionForJenkinsBaseConfigurator() throws Exception {
+        assertThat(validateSchema(convertYamlFileToJson(this, "validJenkinsBaseConfigWithSymbol.yml")),
+            empty());
+    }
+
+    @Test
+    public void validSelfConfigurator() throws Exception {
+        assertThat(
+            validateSchema(convertYamlFileToJson(this, "validSelfConfig.yml")),
+            empty());
+    }
+
+    @Test
+    public void attributesNotFlattenedToTopLevel() throws Exception {
+        assertThat(
+            validateSchema(convertYamlFileToJson(this, "attributesNotFlattenedToTop.yml")),
+            contains("#/tool: extraneous key [acceptLicense] is not permitted"));
```

Comment:
Would we want to make this a bit more generic instead of just validating if the array contains a string